File: Python/Abstraction.py
from abc import ABC,abstractclassmethod, abstractmethod

class Shapes:
    
    @abstractmethod
    def howManySides(self):
        pass
class Triangle(Shapes):
    def howManySides(self):
        print("I have three sides ")
class Square(Shapes):
    def howManySides(self):
        print("I have four sides ")

def mainCallback():
    Tri=Triangle()
    Tri.howManySides()
    # Square
    Sqr=Square()
    Sqr.howManySides()
    if(issubclass(Triangle,Shapes)):
        print("true")
    else:
        print("false")
mainCallback();

File: JavaScript/Abstraction.js
class Shapes {
  howManySides() {
    //not a constructor
    this.sides = null;
  }
}

class Triangle extends Shapes {
  howManySides() {
    this.sides = " Triangle have Three sides";
    console.log(this.sides);
  }
}

class Square extends Shapes {
  howManySides() {
    this.sides = " Square have four sides";
    console.log(this.sides);
  }
}

function mainCallBack() {
  var Tri = new Triangle();
  Tri.howManySides();
  Sqr = new Square();
  Sqr.howManySides();

  if (Tri instanceof Shapes && Sqr instanceof Shapes) {
    console.log("true");
  } else {
    console.log("no");
  }
}
mainCallBack();
